mu-plugins/rest-api/endpoints/class-wporg-plugins-locale-banner-controller.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace WordPressdotorg\MU_Plugins\REST_API;
+
+use WordPressdotorg\Plugin_Directory\Plugin_Directory;
+use function WordPressdotorg\MU_Plugins\Helpers\Locale\{ get_all_locales_with_subdomain, get_all_valid_locales, get_locale_from_header, get_translated_locales };
+
+/**
+ * Plugins_Locale_Banner_Controller
+ */
+class Plugins_Locale_Banner_Controller extends Base_Locale_Banner_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wporg-plugins/v1';
+		$this->rest_base = 'locale-banner';
+	}
+
+	/**
+	 * Validate the plugin slug.
+	 */
+	public function check_slug( $param ) {
+		return is_string( $param ) && $param && Plugin_Directory::get_plugin_post( $param );
+	}
+
+	/**
+	 * Get banner for general plugin directory
+	 */
+	public function get_response( $request ) {
+		if ( ! defined( 'GLOTPRESS_LOCALES_PATH' ) ) {
+			return;
+		}
+
+		require_once GLOTPRESS_LOCALES_PATH;
+
+		$locale_subdomain_assoc = get_all_locales_with_subdomain();
+		$current_locale = get_locale();
+		$current_gp_locale = \GP_Locales::by_field( 'wp_locale', $current_locale );
+
+		// Build a list of WordPress locales which we'll suggest to the user.
+		$suggest_locales = array_values( array_intersect( get_locale_from_header(), get_all_valid_locales() ) );
+
+		$suggestion_links = [];
+		foreach ( $suggest_locales as $locale ) {
+			$language = \GP_Locales::by_field( 'wp_locale', $locale )->native_name;
+			$suggestion_links[ $locale ] = sprintf(
+				'<a href="https://%s.wordpress.org%s">%s</a>',
+				$locale_subdomain_assoc[ $locale ]->subdomain,
+				esc_url( get_site()->path ),
+				$language
+			);
+		}
+
+		$suggest_string = '';
+
+		unset( $suggestion_links[ $current_locale ] );
+
+		if ( ! empty( $suggestion_links ) ) {
+			$output_locale = key( $suggestion_links );
+			switch_to_locale( $output_locale );
+			$suggest_string = sprintf(
+				// translators: %s: List of links to plugin directory in other locales.
+				__( 'The plugin directory is also available in %s.', 'wporg' ),
```

Comment:
I'm wondering if it's worthwhile making these available in the plugin directory translate project via `wporg-plugins` textdomain. It's not impossible to make that happen.

Reply:
I have no strong feelings either way, but I don't know how to do that if it's more complicated than just changing the textdomain.

Reply:
After looking into this, I don't think it's worth it.

It'd require updating `translate/bin/generate-pot.sh` to create a duplicate pot file and then merge them, etc...

The only question I would make instead then is, `would it be worthwhile moving the child-classes of the abstract to be within the relevant directory code bases`, right now, I don't think it's worth it. If translators get confused about the string being missing, we can see about changes later on.

I think this is fine to merge @ryelle.